sycl/test/config/white_list.cpp

```diff
@@ -0,0 +1,60 @@
+// REQUIRES: cpu
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env PRINT_DEVICE_INFO=1 %t.out > %t.conf
+// RUN: env TEST_DEVICE_AVAILABLE=1 env SYCL_CONFIG_FILE_NAME=%t.conf %t.out
+// RUN: env TEST_DEVICE_IS_NOT_AVAILABLE=1 env SYCL_DEVICE_WHITE_LIST="" %t.out
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <cstdlib>
+#include <exception>
+#include <string>
+
+using namespace cl;
+
+int main() {
+
+  // Expected that white list filter is not set
+  if (getenv("PRINT_DEVICE_INFO")) {
+    for (const sycl::platform &Plt : sycl::platform::get_platforms())
+      if (!Plt.is_host()) {
+        const sycl::device Dev = Plt.get_devices().at(0);
+        std::string DevName = Dev.get_info<sycl::info::device::name>();
+        const std::string DevVer =
+            Dev.get_info<sycl::info::device::driver_version>();
+        // As device name string will be used as regexp pattern, we need to
+        // get rid of symbols that can be treated in a special way.
+        // Replace common special symbols with '.' which matches to any sybmol
+        for (char &Sym : DevName) {
```

Comment:
What about using algorithms instead?

Reply:
Ok, will replace with:
```
std::replace_if(DevName.begin(), DevName.end(),
                       [](const char Sym) { return '(' == Sym || ')' == Sym; }, '.');
```
Speaking honestly, I think loop code is more readable.

Reply:
I was thinking to a simpler:
```
std::replace(DevName.begin(), DevName.end(), '(', '.');
std::replace(DevName.begin(), DevName.end(), ')', '.');
```
since everything is in the cache, it should be fast.

Reply:
I don't think we should sacrifice performance in order to make C++ code more readable...

Reply:
I think it is very subjective.
```
for (char &Sym : DevName) {
  if (')' == Sym || '(' == Sym)
    Sym = '.';
}
```
IMO this code is more readable than:
```
std::replace(DevName.begin(), DevName.end(), '(', '.');
std::replace(DevName.begin(), DevName.end(), ')', '.');
```